### ISSUE TITLE
1.4.6+ duplicate record found in sqlite

### ISSUE THREAD

**Issue by youxq** [fluent/fluent-bit#2576]
## Bug Report

**Describe the bug**
1.4.6+ duplicate record found in sqlite

**To Reproduce**
- Rubular link if applicable:
- Steps to reproduce the problem:
```
# input config
[INPUT]
    Name tail
    Path /data/k8s/log/proxy*/log/api-platform-proxy/proxy*.log
    Path_Key path
    Skip_Long_Lines On
    DB /data/logs.db
    Tag 5319095990785211753_api_platform_proxy

# target file list
[root@host flb]# ll /data/k8s/log/proxy*/log/api-platform-proxy/proxy*.log
-rw-r--r-- 1 root root 104860085 Sep 16 20:29 /data/k8s/log/proxy/log/api-platform-proxy/proxy_2020-09-16_2.log
-rw-r--r-- 1 root root  33099285 Sep 16 23:59 /data/k8s/log/proxy/log/api-platform-proxy/proxy_2020-09-16_3.log
-rw-r--r-- 1 root root 104859923 Sep 17 09:30 /data/k8s/log/proxy/log/api-platform-proxy/proxy_2020-09-17_1.log
-rw-r--r-- 1 root root 104860681 Sep 17 20:10 /data/k8s/log/proxy/log/api-platform-proxy/proxy_2020-09-17_2.log
-rw-r--r-- 1 root root  37073945 Sep 17 23:59 /data/k8s/log/proxy/log/api-platform-proxy/proxy_2020-09-17_3.log
-rw-r--r-- 1 root root 104857744 Sep 18 09:34 /data/k8s/log/proxy/log/api-platform-proxy/proxy_2020-09-18_1.log
-rw-r--r-- 1 root root 104866722 Sep 18 20:11 /data/k8s/log/proxy/log/api-platform-proxy/proxy_2020-09-18_2.log
-rw-r--r-- 1 root root  36162878 Sep 18 23:59 /data/k8s/log/proxy/log/api-platform-proxy/proxy_2020-09-18_3.log
-rw-r--r-- 1 root root 104860413 Sep 19 09:28 /data/k8s/log/proxy/log/api-platform-proxy/proxy_2020-09-19_1.log
-rw-r--r-- 1 root root 104858600 Sep 19 20:09 /data/k8s/log/proxy/log/api-platform-proxy/proxy_2020-09-19_2.log
-rw-r--r-- 1 root root  36523332 Sep 19 23:59 /data/k8s/log/proxy/log/api-platform-proxy/proxy_2020-09-19_3.log
-rw-r--r-- 1 root root 104857607 Sep 20 10:13 /data/k8s/log/proxy/log/api-platform-proxy/proxy_2020-09-20_1.log
-rw-r--r-- 1 root root 104857667 Sep 20 21:08 /data/k8s/log/proxy/log/api-platform-proxy/proxy_2020-09-20_2.log
-rw-r--r-- 1 root root  27576909 Sep 20 23:59 /data/k8s/log/proxy/log/api-platform-proxy/proxy_2020-09-20_3.log
-rw-r--r-- 1 root root 104863565 Sep 21 09:32 /data/k8s/log/proxy/log/api-platform-proxy/proxy_2020-09-21_1.log
-rw-r--r-- 1 root root 104866090 Sep 21 20:06 /data/k8s/log/proxy/log/api-platform-proxy/proxy_2020-09-21_2.log
-rw-r--r-- 1 root root  17097106 Sep 21 21:48 /data/k8s/log/proxy/log/api-platform-proxy/proxy_2020-09-21_3.log
-rw-r--r-- 1 root root 104857630 Sep 22 09:49 /data/k8s/log/proxy/log/api-platform-proxy/proxy_2020-09-22_1.log
-rw-r--r-- 1 root root  88810985 Sep 23 19:33 /data/k8s/log/proxy/log/api-platform-proxy/proxy.log

# trace record in sqlite
174|/data/k8s/log/proxy/log/api-platform-proxy/proxy.log|2602846|11124377538|1600860409|0
175|/data/k8s/log/proxy/log/api-platform-proxy/proxy.log|2549858|11124377538|1600860409|0
176|/data/k8s/log/proxy/log/api-platform-proxy/proxy.log|2555681|11124377538|1600860409|0
177|/data/k8s/log/proxy/log/api-platform-proxy/proxy.log|2818765|11124377538|1600860409|0
178|/data/k8s/log/proxy/log/api-platform-proxy/proxy.log|2540686|11124377538|1600860409|0
179|/data/k8s/log/proxy/log/api-platform-proxy/proxy.log|2759291|11124377538|1600860409|0
180|/data/k8s/log/proxy/log/api-platform-proxy/proxy.log|2745840|11124377538|1600860409|0
181|/data/k8s/log/proxy/log/api-platform-proxy/proxy.log|2559475|11124377538|1600860409|0
182|/data/k8s/log/proxy/log/api-platform-proxy/proxy.log|2829909|11124377538|1600860409|0
183|/data/k8s/log/proxy/log/api-platform-proxy/proxy.log|2800861|11124377538|1600860409|0
184|/data/k8s/log/proxy/log/api-platform-proxy/proxy.log|2574409|11124377538|1600860409|0
185|/data/k8s/log/proxy/log/api-platform-proxy/proxy.log|2744466|11124377538|1600860409|0
186|/data/k8s/log/proxy/log/api-platform-proxy/proxy.log|2747425|11124377538|1600860409|0
187|/data/k8s/log/proxy/log/api-platform-proxy/proxy.log|2766320|11124377538|1600860409|0
188|/data/k8s/log/proxy/log/api-platform-proxy/proxy.log|2740079|11124377538|1600860409|0
189|/data/k8s/log/proxy/log/api-platform-proxy/proxy.log|2723852|11124377538|1600860409|0
190|/data/k8s/log/proxy/log/api-platform-proxy/proxy.log|2493285|11124377538|1600860409|0
191|/data/k8s/log/proxy/log/api-platform-proxy/proxy.log|2705592|11124377538|1600860409|0
192|/data/k8s/log/proxy/log/api-platform-proxy/proxy.log|2731726|11124377538|1600860409|0
```

**Expected behavior**

**Screenshots**

**Your Environment**

* Version used: 1.4.6 & 1.5.6 td-agent-bit
* Configuration:
* Environment name and version (e.g. Kubernetes? What version?): 
* Server type and version:
* Operating System and version: centos7.2
* Filters and plugins:

**Additional context**

When I use td-agent-bit version 1.4.5, it's correct, but failed with using 1.4.6+
```
# sqlite record with td-agent-bit 1.4.5
174|/data/k8s/log/proxy/log/api-platform-proxy/proxy.log|1322285|11124377538|1600861197|0
175|/data/k8s/log/proxy/log/api-platform-proxy/proxy_2020-09-16_2.log|1238391|10801490958|1600861197|0
176|/data/k8s/log/proxy/log/api-platform-proxy/proxy_2020-09-16_3.log|1259823|10793348939|1600861197|0
177|/data/k8s/log/proxy/log/api-platform-proxy/proxy_2020-09-17_1.log|1397934|11099022381|1600861197|0
178|/data/k8s/log/proxy/log/api-platform-proxy/proxy_2020-09-17_2.log|1279292|10801633927|1600861197|0
179|/data/k8s/log/proxy/log/api-platform-proxy/proxy_2020-09-17_3.log|1395155|10801373151|1600861197|0
180|/data/k8s/log/proxy/log/api-platform-proxy/proxy_2020-09-18_1.log|1378953|11101777903|1600861197|0
181|/data/k8s/log/proxy/log/api-platform-proxy/proxy_2020-09-18_2.log|1318312|11101464709|1600861197|0
182|/data/k8s/log/proxy/log/api-platform-proxy/proxy_2020-09-18_3.log|1409282|11151921210|1600861197|0
183|/data/k8s/log/proxy/log/api-platform-proxy/proxy_2020-09-19_1.log|1393167|10801698604|1600861197|0
184|/data/k8s/log/proxy/log/api-platform-proxy/proxy_2020-09-19_2.log|1267343|10801628722|1600861197|0
185|/data/k8s/log/proxy/log/api-platform-proxy/proxy_2020-09-19_3.log|1367611|11134686668|1600861197|0
186|/data/k8s/log/proxy/log/api-platform-proxy/proxy_2020-09-20_1.log|1355527|11116650915|1600861197|0
187|/data/k8s/log/proxy/log/api-platform-proxy/proxy_2020-09-20_2.log|1366988|11140526047|1600861197|0
188|/data/k8s/log/proxy/log/api-platform-proxy/proxy_2020-09-20_3.log|1371209|11115991224|1600861197|0
189|/data/k8s/log/proxy/log/api-platform-proxy/proxy_2020-09-21_1.log|1369087|10801086936|1600861197|0
190|/data/k8s/log/proxy/log/api-platform-proxy/proxy_2020-09-21_2.log|1251774|10801218203|1600861197|0
191|/data/k8s/log/proxy/log/api-platform-proxy/proxy_2020-09-21_3.log|1337684|10791332679|1600861197|0
192|/data/k8s/log/proxy/log/api-platform-proxy/proxy_2020-09-22_1.log|1354927|11142163916|1600861197|0
```

**Comment by sbvitok** [comment]
fluent-bit version 1.5.7

```
[INPUT]
    Name              tail
    Path              {{ .Values.volumes.pods.mountPath }}/*/volumes/kubernetes.io~empty-dir/log/tp_*.log
    Tag               tp.*
    Refresh_Interval  5
    Mem_Buf_Limit     256MB
    Skip_Long_Lines   On
    DB                {{ .Values.volumes.taildb.mountPath }}/tail-tp-state.db

```
sql

```
[root@vkp-kube-node-1.i ~]# sqlite3 /var/lib/fluent-bit/tail-tp-state.db 
SQLite version 3.7.17 2013-05-20 00:56:22
Enter ".help" for instructions
Enter SQL statements terminated with a ";"
sqlite> .tables
in_tail_files
sqlite> select * from in_tail_files;
1|/var/lib/kubelet/pods/63e0fb6a-2458-4db2-b3fe-c0b8b8504b56/volumes/kubernetes.io~empty-dir/log/tp_master.log|1388|540214786|1601218818|0
2|/var/lib/kubelet/pods/63e0fb6a-2458-4db2-b3fe-c0b8b8504b56/volumes/kubernetes.io~empty-dir/log/tp_master.log|4322548|540214786|1601218818|0
3|/var/lib/kubelet/pods/63e0fb6a-2458-4db2-b3fe-c0b8b8504b56/volumes/kubernetes.io~empty-dir/log/tp_master.log|2169097|540214786|1601218818|0
4|/var/lib/kubelet/pods/63e0fb6a-2458-4db2-b3fe-c0b8b8504b56/volumes/kubernetes.io~empty-dir/log/tp_master.log|4338458|540214786|1601218818|0
5|/var/lib/kubelet/pods/63e0fb6a-2458-4db2-b3fe-c0b8b8504b56/volumes/kubernetes.io~empty-dir/log/tp_master.log|45295261|540214786|1601218818|0
6|/var/lib/kubelet/pods/63e0fb6a-2458-4db2-b3fe-c0b8b8504b56/volumes/kubernetes.io~empty-dir/log/tp_master.log|2153718|540214786|1601218818|0
7|/var/lib/kubelet/pods/63e0fb6a-2458-4db2-b3fe-c0b8b8504b56/volumes/kubernetes.io~empty-dir/log/tp_master.log|12570744|540214786|1601218818|0
8|/var/lib/kubelet/pods/63e0fb6a-2458-4db2-b3fe-c0b8b8504b56/volumes/kubernetes.io~empty-dir/log/tp_slot.100.log|4352228|540214788|1601220312|0
9|/var/lib/kubelet/pods/63e0fb6a-2458-4db2-b3fe-c0b8b8504b56/volumes/kubernetes.io~empty-dir/log/tp_slot.100.log|2183993|540214788|1601220312|0
10|/var/lib/kubelet/pods/63e0fb6a-2458-4db2-b3fe-c0b8b8504b56/volumes/kubernetes.io~empty-dir/log/tp_slot.100.log|4368250|540214788|1601220312|0
11|/var/lib/kubelet/pods/63e0fb6a-2458-4db2-b3fe-c0b8b8504b56/volumes/kubernetes.io~empty-dir/log/tp_slot.100.log|45605741|540214788|1601220312|0
12|/var/lib/kubelet/pods/63e0fb6a-2458-4db2-b3fe-c0b8b8504b56/volumes/kubernetes.io~empty-dir/log/tp_slot.100.log|2168502|540214788|1601220312|0
13|/var/lib/kubelet/pods/63e0fb6a-2458-4db2-b3fe-c0b8b8504b56/volumes/kubernetes.io~empty-dir/log/tp_slot.100.log|12657444|540214788|1601220312|0
sqlite> 
[root@vkp-kube-node-1.i ~]# ls /var/lib/kubelet/pods/63e0fb6a-2458-4db2-b3fe-c0b8b8504b56/volumes/kubernetes.io~empty-dir/log/ -lc
total 145688
-rw-r--r-- 1 root root     1388 Sep 25 21:06 tp_master.log
-rw-r--r-- 1 root root  4562903 Sep 27 20:55 tp_slot.100.log
-rw-r--r-- 1 root root  2289728 Sep 27 20:55 tp_slot.200.log
-rw-r--r-- 1 root root  4579720 Sep 27 20:55 tp_slot.300.log
-rw-r--r-- 1 root root 47832845 Sep 27 20:55 tp_slot.400.log
-rw-r--r-- 1 root root  2273442 Sep 27 20:55 tp_slot.500.log
-rw-r--r-- 1 root root 13270124 Sep 27 20:55 tp_worker.log
[root@vkp-kube-node-1.i ~]# ls -lc /var/lib/fluent-bit/tail-containers-state.db 
-rw-r--r-- 1 root root 32768 Sep 27 20:59 /var/lib/fluent-bit/tail-containers-state.db
[root@vkp-kube-node-1.i ~]#

```
Same with tail /var/log/containers/*.log but more output for display.
And if I restart fluent-bit, the files what not in db will by resend again.

**Comment by sbvitok** [comment]
fresh install:
```
[root@vkp-kube-node-1.i ~]# sqlite3 /var/lib/fluent-bit/tail-containers-state.db "select * from in_tail_files"                                                                                
1|/var/log/containers/calico-node-x2jh6_kube-system_calico-node-ce4c1787351b231ca64521dfa6d09f4d4e33e3d0d047e7b0314d6586593501bb.log|4472071|806467172|1601232882|0
2|/var/log/containers/calico-node-x2jh6_kube-system_calico-node-ce4c1787351b231ca64521dfa6d09f4d4e33e3d0d047e7b0314d6586593501bb.log|109|806467172|1601232882|0
3|/var/log/containers/calico-node-x2jh6_kube-system_calico-node-ce4c1787351b231ca64521dfa6d09f4d4e33e3d0d047e7b0314d6586593501bb.log|3830|806467172|1601232882|0
4|/var/log/containers/calico-node-x2jh6_kube-system_calico-node-ce4c1787351b231ca64521dfa6d09f4d4e33e3d0d047e7b0314d6586593501bb.log|744|806467172|1601232882|0
5|/var/log/containers/calico-node-x2jh6_kube-system_calico-node-ce4c1787351b231ca64521dfa6d09f4d4e33e3d0d047e7b0314d6586593501bb.log|755700|806467172|1601232882|0
6|/var/log/containers/calico-node-x2jh6_kube-system_calico-node-ce4c1787351b231ca64521dfa6d09f4d4e33e3d0d047e7b0314d6586593501bb.log|11030|806467172|1601232882|0
7|/var/log/containers/calico-node-x2jh6_kube-system_calico-node-ce4c1787351b231ca64521dfa6d09f4d4e33e3d0d047e7b0314d6586593501bb.log|4481141|806467172|1601232882|0
8|/var/log/containers/calico-node-x2jh6_kube-system_calico-node-ce4c1787351b231ca64521dfa6d09f4d4e33e3d0d047e7b0314d6586593501bb.log|12005|806467172|1601232882|0
9|/var/log/containers/calico-node-x2jh6_kube-system_calico-node-ce4c1787351b231ca64521dfa6d09f4d4e33e3d0d047e7b0314d6586593501bb.log|7923|806467172|1601232882|0
10|/var/log/containers/calico-node-x2jh6_kube-system_calico-node-ce4c1787351b231ca64521dfa6d09f4d4e33e3d0d047e7b0314d6586593501bb.log|4456655|806467172|1601232882|0
11|/var/log/containers/calico-node-x2jh6_kube-system_calico-node-ce4c1787351b231ca64521dfa6d09f4d4e33e3d0d047e7b0314d6586593501bb.log|4464431|806467172|1601232882|0
12|/var/log/containers/calico-node-x2jh6_kube-system_calico-node-ce4c1787351b231ca64521dfa6d09f4d4e33e3d0d047e7b0314d6586593501bb.log|10496|806467172|1601232882|0
13|/var/log/containers/calico-node-x2jh6_kube-system_calico-node-ce4c1787351b231ca64521dfa6d09f4d4e33e3d0d047e7b0314d6586593501bb.log|4463830|806467172|1601232882|0
14|/var/log/containers/calico-node-x2jh6_kube-system_calico-node-ce4c1787351b231ca64521dfa6d09f4d4e33e3d0d047e7b0314d6586593501bb.log|18307|806467172|1601232882|0
15|/var/log/containers/calico-node-x2jh6_kube-system_calico-node-ce4c1787351b231ca64521dfa6d09f4d4e33e3d0d047e7b0314d6586593501bb.log|13271|806467172|1601232882|0
16|/var/log/containers/calico-node-x2jh6_kube-system_calico-node-ce4c1787351b231ca64521dfa6d09f4d4e33e3d0d047e7b0314d6586593501bb.log|12004|806467172|1601232882|0
17|/var/log/containers/calico-node-x2jh6_kube-system_calico-node-ce4c1787351b231ca64521dfa6d09f4d4e33e3d0d047e7b0314d6586593501bb.log|28477|806467172|1601232882|0
18|/var/log/containers/calico-node-x2jh6_kube-system_calico-node-ce4c1787351b231ca64521dfa6d09f4d4e33e3d0d047e7b0314d6586593501bb.log|57905|806467172|1601232882|0
19|/var/log/containers/calico-node-x2jh6_kube-system_calico-node-ce4c1787351b231ca64521dfa6d09f4d4e33e3d0d047e7b0314d6586593501bb.log|11887|806467172|1601232882|0
20|/var/log/containers/calico-node-x2jh6_kube-system_calico-node-ce4c1787351b231ca64521dfa6d09f4d4e33e3d0d047e7b0314d6586593501bb.log|4915|806467172|1601232882|0
21|/var/log/containers/calico-node-x2jh6_kube-system_calico-node-ce4c1787351b231ca64521dfa6d09f4d4e33e3d0d047e7b0314d6586593501bb.log|6510|806467172|1601232882|0
22|/var/log/containers/calico-node-x2jh6_kube-system_calico-node-ce4c1787351b231ca64521dfa6d09f4d4e33e3d0d047e7b0314d6586593501bb.log|2661|806467172|1601232882|0
23|/var/log/containers/calico-node-x2jh6_kube-system_calico-node-ce4c1787351b231ca64521dfa6d09f4d4e33e3d0d047e7b0314d6586593501bb.log|4479110|806467172|1601232882|0
24|/var/log/containers/calico-node-x2jh6_kube-system_calico-node-ce4c1787351b231ca64521dfa6d09f4d4e33e3d0d047e7b0314d6586593501bb.log|12005|806467172|1601232882|0
25|/var/log/containers/calico-node-x2jh6_kube-system_calico-node-ce4c1787351b231ca64521dfa6d09f4d4e33e3d0d047e7b0314d6586593501bb.log|1808|806467172|1601232882|0
26|/var/log/containers/calico-node-x2jh6_kube-system_calico-node-ce4c1787351b231ca64521dfa6d09f4d4e33e3d0d047e7b0314d6586593501bb.log|4418163|806467172|1601232882|0
27|/var/log/containers/calico-node-x2jh6_kube-system_calico-node-ce4c1787351b231ca64521dfa6d09f4d4e33e3d0d047e7b0314d6586593501bb.log|12004|806467172|1601232882|0
28|/var/log/containers/calico-node-x2jh6_kube-system_calico-node-ce4c1787351b231ca64521dfa6d09f4d4e33e3d0d047e7b0314d6586593501bb.log|1793|806467172|1601232882|0
29|/var/log/containers/calico-node-x2jh6_kube-system_calico-node-ce4c1787351b231ca64521dfa6d09f4d4e33e3d0d047e7b0314d6586593501bb.log|1055|806467172|1601232882|0
30|/var/log/containers/calico-node-x2jh6_kube-system_calico-node-ce4c1787351b231ca64521dfa6d09f4d4e33e3d0d047e7b0314d6586593501bb.log|5972|806467172|1601232882|0
31|/var/log/containers/calico-node-x2jh6_kube-system_calico-node-ce4c1787351b231ca64521dfa6d09f4d4e33e3d0d047e7b0314d6586593501bb.log|0|806467172|1601232882|0
32|/var/log/containers/calico-node-x2jh6_kube-system_calico-node-ce4c1787351b231ca64521dfa6d09f4d4e33e3d0d047e7b0314d6586593501bb.log|29109|806467172|1601232882|0
33|/var/log/containers/calico-node-x2jh6_kube-system_calico-node-ce4c1787351b231ca64521dfa6d09f4d4e33e3d0d047e7b0314d6586593501bb.log|5699|806467172|1601232882|0
34|/var/log/containers/calico-node-x2jh6_kube-system_calico-node-ce4c1787351b231ca64521dfa6d09f4d4e33e3d0d047e7b0314d6586593501bb.log|228|806467172|1601232882|0
[root@vkp-kube-node-1.i ~]# ls /var/log/containers/|wc -l
34
[root@vkp-kube-node-1.i ~]#
```

**Comment by sbvitok** [comment]
but I see problem only on 1.5.5, 1.5.6, 1.5.7 versions
1.5.4 - work for my.

**Comment by edsiper** [comment]
I think this is totally related to the following fix: 8a5309a2f29902f5a2031113ecaa68fbdb902adb . 

I've found a similar issue and if it worked properly on 1.5.4 is because that missing bindings reset.
